doc/src/extras.rst

```diff
@@ -141,6 +141,262 @@ Logging cursor
+
+           On the other hand, it is not recommended to send a confirmation
+           after every processed message, since that will put an unnecessary
+           load on network and the server.  A possible strategy is to confirm
+           after every COMMIT message.
+
+    .. method:: stop_replication()
+
+        In non-asynchronous connection, when called from the ``write()``
+        method, tell the code in `~start_replication` to break out of the
+        endless loop and return.
+
+    .. method:: send_replication_feedback(write_lsn=0, flush_lsn=0, apply_lsn=0, reply=False)
+
+        :param write_lsn: a LSN position up to which the client has written the data locally
+        :param flush_lsn: a LSN position up to which the client has stored the
+                          data reliably (the server is allowed to discard all
+                          and every data that predates this LSN)
+        :param apply_lsn: a LSN position up to which the warm standby server
+                          has applied the changes (physical replication
+                          master-slave protocol only)
+        :param reply: request the server to send back a keepalive message immediately
+
+        Use this method to report to the server that all messages up to a
+        certain LSN position have been stored and may be discarded.
+
+        This method can also be called with all default parameters' values to
+        send a keepalive message to the server.
+
+        In case of asynchronous connection, if the feedback message cannot be
+        sent at the moment, remembers the passed LSN positions for a later
+        hopefully successful call or call to `~flush_replication_feedback()`.
+
+    .. method:: flush_replication_feedback(reply=False)
+
+        :param reply: request the server to send back a keepalive message immediately
+
+        This method tries to flush the latest replication feedback message
+        that `~send_replication_feedback()` was trying to send, if any.
+
+    Low-level methods for asynchronous connection operation.
+
+    With the non-asynchronous connection, a single call to
+    `~start_replication()` handles all the complexity, but at times it might
+    be beneficial to use low-level interface for better control, in particular
+    to `~select.select()` on multiple sockets.  The following methods are
+    provided for asynchronous operation:
+
+    .. method:: read_replication_message(decode=True)
+
+        :param decode: a flag indicating that unicode conversion should be
+                       performed on the data received from the server
+
+        This method should be used in a loop with asynchronous connections
+        after calling `~start_replication()` once.
+
+        It tries to read the next message from the server, without blocking
+        and returns an instance of `~ReplicationMessage` or *None*, in case
+        there are no more data messages from the server at the moment.
```

Comment:
Should make it clear that a whole message is returned; partial message data is buffered then returned once a complete message is available.